lib/traject/adjust_cardinality.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Traject
+  # Given a data structure from traject, transform it into a valid IR by changing
+  # some values from arrays to scalars
+  class AdjustCardinality
+    def self.call(attributes)
+      new(attributes).call
+    end
+
+    def initialize(attributes)
+      @source = attributes
+    end
+
+    attr_reader :source
+
+    def call
+      flatten_top_level(source)
+    end
+
+    def flatten_top_level(attributes)
+      flatten = %w(id modsxml access_facet building_facet druid url_fulltext display_type collection_type)
+      flatten += %w(author_1xx_search)
```

Comment:
Should we be using `<<` / `concat` to prevent us from unnecessarily creating a new object each time?

Reply:
I think a next step for this code is to stop adjusting cardinality; Solr doesn't care, so this is only to create perfect parity with the current indexing.

Reply:
I'm not sure I understand what that has to do with using `+=` over `<<`/`concat` (it's my understanding the latter is more performant in not creating additional objects while exhibiting the same resulting data structure), but I also don't think that should hold this PR up.

Reply:
I was just suggesting that, ideally, the whole thing will go away soon; I see it as just a hack to ensure complete parity with the old output. Since you seem concerned about it, I've replaced it with a single array without any concat'ing.